src/main/java/hudson/plugins/jira/JiraCreateIssueNotifier.java

```diff
@@ -134,25 +137,17 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
      */
     private Issue createJiraIssue(AbstractBuild<?, ?> build, String filename) throws IOException, InterruptedException {
 
-        EnvVars environmentVariable = build.getEnvironment(TaskListener.NULL);
-
-        String buildURL = environmentVariable.get("BUILD_URL");
-        String buildNumber = environmentVariable.get("BUILD_NUMBER");
-        String jobName = environmentVariable.get("JOB_NAME");
-        String jenkinsURL = Jenkins.getInstance().getRootUrl();
+        EnvVars vars = build.getEnvironment(TaskListener.NULL);
 
+        String summary = String.format("Build %s failed", getBuildName(vars));
         String checkDescription = (this.testDescription.equals("")) ? "No description is provided" : this.testDescription;
```

Comment:
testDescription is the preference selected in the UI, so overwriting it here will probably overwrite the plugin configuration during serialization? That's my wild guess, but I might be completely wrong.. I've assumed that also this is the reason the fields like build, job, etc.. are not class fields but local variables, because they would be treated as configuration unless transient?